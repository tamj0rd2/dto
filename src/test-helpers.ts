import { resolve } from 'path'
import { readdirSync } from 'fs'

export const runTests = (testFolder: string, testPath: string) => {
  const fixturePrefix = testPath.replace(`${testFolder}/`, '').replace('.spec.ts', '')
  const files = readdirSync(resolve(process.cwd(), 'src', '__fixtures__', fixturePrefix))

  files.forEach((fileName) => {
    test(fileName, () => {
      expect(`${fixturePrefix}/${fileName}`).toMatchDiagnosticsSnapshot()
    })
  })
}
