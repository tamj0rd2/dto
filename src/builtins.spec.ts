import { runTests } from './test-helpers'

runTests(__dirname, __filename)
