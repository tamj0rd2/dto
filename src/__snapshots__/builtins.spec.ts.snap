// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date.ts 1`] = `
Object {
  "errorCount": 4,
  "fileName": "builtins/date.ts",
  "messages": Array [
    "builtins/date.ts:7(14) Type 'Date' is not assignable to type 'string'.",
    "builtins/date.ts:9(14) Type 'null' is not assignable to type 'string'.",
    "builtins/date.ts:11(14) Type 'undefined' is not assignable to type 'string'.",
    "builtins/date.ts:13(14) Type '{ bad: string; }' is not assignable to type 'string'.",
  ],
}
`;

exports[`optional-date.ts 1`] = `
Object {
  "errorCount": 3,
  "fileName": "builtins/optional-date.ts",
  "messages": Array [
    "builtins/optional-date.ts:9(14) Type 'Date' is not assignable to type 'string'.",
    "builtins/optional-date.ts:11(14) Type 'undefined' is not assignable to type 'string | null'.",
    "builtins/optional-date.ts:13(14) Type '{ bad: string; }' is not assignable to type 'string'.",
  ],
}
`;

exports[`optional-set.ts 1`] = `
Object {
  "errorCount": 5,
  "fileName": "builtins/optional-set.ts",
  "messages": Array [
    "builtins/optional-set.ts:9(14) Type 'Set<number>' is not assignable to type 'number[]'.",
    "builtins/optional-set.ts:11(14) Type 'undefined' is not assignable to type 'number[] | null'.",
    "builtins/optional-set.ts:13(14) Type '\\"this is not good\\"' is not assignable to type 'number[] | null'.",
    "builtins/optional-set.ts:15(14) Type '192843' is not assignable to type 'number[] | null'.",
    "builtins/optional-set.ts:17(47) Type '{ bad: string; }' is not assignable to type 'number[]'.
  Object literal may only specify known properties, and 'bad' does not exist in type 'number[]'.",
  ],
}
`;

exports[`set.ts 1`] = `
Object {
  "errorCount": 6,
  "fileName": "builtins/set.ts",
  "messages": Array [
    "builtins/set.ts:7(14) Type 'Set<number>' is missing the following properties from type 'number[]': length, pop, push, concat, and 23 more.",
    "builtins/set.ts:9(14) Type 'null' is not assignable to type 'number[]'.",
    "builtins/set.ts:11(14) Type 'undefined' is not assignable to type 'number[]'.",
    "builtins/set.ts:13(14) Type 'string' is not assignable to type 'number[]'.",
    "builtins/set.ts:15(14) Type 'number' is not assignable to type 'number[]'.",
    "builtins/set.ts:17(39) Type '{ bad: string; }' is not assignable to type 'number[]'.
  Object literal may only specify known properties, and 'bad' does not exist in type 'number[]'.",
  ],
}
`;
